src/sync/sync_manager.cpp

```diff
@@ -344,20 +348,27 @@ std::shared_ptr<SyncUser> SyncManager::get_current_user() const
     
     auto is_active_user = [](auto& el) { return el.second->state() == SyncUser::State::Active; };
     auto it = std::find_if(m_users.begin(), m_users.end(), is_active_user);
-    if (it == m_users.end()) {
+    if (it == m_users.end())
         return nullptr;
-    }
-    if (std::find_if(std::next(it), m_users.end(), is_active_user) != m_users.end()) {
+
+    if (std::find_if(std::next(it), m_users.end(), is_active_user) != m_users.end())
         throw std::logic_error("Current user is not valid if more that one valid, logged-in user exists.");
-    }
+
     return it->second;
 }
 
-std::string SyncManager::path_for_realm(const std::string& user_identity, const std::string& raw_realm_url) const
+std::string SyncManager::path_for_realm(const SyncUser& user, const std::string& raw_realm_url) const
 {
     std::lock_guard<std::mutex> lock(m_file_system_mutex);
     REALM_ASSERT(m_file_manager);
-    return m_file_manager->path(user_identity, raw_realm_url);
+    const auto user_identity = user.identity();
+    const auto user_local_identity = user.local_identity();
+    std::string local_identity = user_local_identity.value_or(user_identity);
```

Comment:
`user_identity`, `user_local_identity` and `local_identity` should all be references to avoid unnecessary copying.